### DISCUSSION TITLE
Get developer version of debugger?

### DISCUSSION THREAD

**Discussion by yanik-recke** [Vector35/binaryninja-api#3556] [General] [answered]
Is there a way to get the most recent developer version of the debugger? I am running into issue 150 (https://github.com/Vector35/debugger/issues/150) but it says solved so I think it is just because I do not have the most recent version?

Best regards

**Comment by psifertex** [comment, marked as answer]
Sorry for not replying earlier but the latest version of the debugger is now built and included in the product by default. Just switch to the dev branch to make sure you're on the [latest dev build](https://docs.binary.ninja/guide/index.html#updates).

**Comment by xusheng6** [comment]
As @psifertex mentioned, using the dev build of Binary Ninja should give you the latest or near latest dev build of the debugger. And that is likely to include the bug fixes you are looking for. 

However, there might also be a small delay between a bug is fixed on the dev branch of the debugger and the bug fix is available in the latest dev version of the BN. Because I need to push a BN dev build pointing to the latest dev of the debugger. There is a plan to publish the dev build of the debugger itself (https://github.com/Vector35/debugger/issues/369), which you can download and use without updating the binary ninja. That will serve as a way to try out the cutting-edge development progress of the debugger.

Regarding your particular issue, I see that issue (https://github.com/Vector35/debugger/issues/150) is closed on Jun 23 and I commented "Fixed in BN 3.1.3529-dev", which is some four months before this issue is created. If you are running a build newer than 3.1.3529, then it should have been fixed. I suspect you are affected by a different issue other than the general failure to debug x86 .exe. As such, I suggest you file a bug report at https://github.com/Vector35/debugger and provide relevant information, so I can troubleshoot and see what is happening. 

Also, do you happen to be unable to debug a particular x86 .exe? Or any x86 .exe?

**Comment by yanik-recke** [comment]
Thanks a lot for the reply. Everything worked when I switched to the dev branch, I just had trouble finding the setting for it but after asking on your slack, someone helped me.

Thanks again.

Sent from my iPhone

On 29. Nov 2022, at 03:52, xusheng ***@***.***> wrote:

﻿

As @psifertex<https://github.com/psifertex> mentioned, using the dev build of Binary Ninja should give you the latest or near latest dev build of the debugger. And that is likely to include the bug fixes you are looking for.

However, there might also be a small delay between a bug is fixed on the dev branch of the debugger and the bug fix is available in the latest dev version of the BN. Because I need to push a BN dev build pointing to the latest dev of the debugger. There is a plan to publish the dev build of the debugger itself (Vector35/debugger#369<https://github.com/Vector35/debugger/issues/369>), which you can download and use without updating the binary ninja. That will serve as a way to try out the cutting-edge development progress of the debugger.

Regarding your particular issue, I see that issue (Vector35/debugger#150<https://github.com/Vector35/debugger/issues/150>) is closed on Jun 23 and I commented "Fixed in BN 3.1.3529-dev", which is some four months before this issue is created. If you are running a build newer than 3.1.3529, then it should have been fixed. I suspect you are affected by a different issue other than the general failure to debug x86 .exe.
